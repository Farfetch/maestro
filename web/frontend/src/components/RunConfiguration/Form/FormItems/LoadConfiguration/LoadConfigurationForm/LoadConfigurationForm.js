import { MinusCircleOutlined, PlusOutlined } from "@ant-design/icons";
import { Button, Col, Form, InputNumber, Row } from "antd";

const formItemLayout = {
  labelCol: { span: 0 },
  wrapperCol: { span: 26 },
  gutter: 0
};

const LoadConfigurationForm = ({ onInputChange, onFieldRemove }) => (
  <>
    <Form.List name="loadProfile">
      {(fields, { add, remove }) => (
        <>
          {fields.length > 0 ? (
            <Row gutter={[8, 0]}>
              <Col flex="auto">
                <Row gutter={[24, 0]}>
                  <Col span={8}>Start RPS</Col>
                  <Col span={8}>End RPS</Col>
                  <Col span={8}>Duration</Col>
                </Row>
              </Col>
              <Col flex="20px"></Col>
            </Row>
          ) : null}
          {fields.map((field, index) => (
            <Row key={field.key} align="middle" gutter={[6, 0]}>
              <Col flex="auto">
                <Row gutter={[16, 0]}>
                  <Col span={8}>
                    <Form.Item
                      {...field}
                      name={[field.name, "start"]}
                      rules={[{ required: true, message: "Missing Start RPS" }]}
                      {...formItemLayout}
                    >
                      <InputNumber
                        style={{ width: "100%" }}
                        min={1}
                        onChange={(value) =>
                          onInputChange(index, "start", value)
                        }
                      />
                    </Form.Item>
                  </Col>
                  <Col span={8}>
                    <Form.Item
                      {...field}
                      name={[field.name, "end"]}
                      rules={[{ required: true, message: "Missing End RPS" }]}
                      {...formItemLayout}
                    >
                      <InputNumber
                        style={{ width: "100%" }}
                        min={1}
                        onChange={(value) => onInputChange(index, "end", value)}
                      />
                    </Form.Item>
                  </Col>
                  <Col span={8}>
                    <Form.Item
                      {...field}
                      name={[field.name, "duration"]}
                      rules={[{ required: true, message: "Missing Duration" }]}
                      {...formItemLayout}
                    >
                      <InputNumber
                        style={{ width: "100%" }}
                        min={1}
                        onChange={(value) =>
                          onInputChange(index, "duration", value)
                        }
                      />
                    </Form.Item>
                  </Col>
                </Row>
              </Col>
              <Col flex="20px" style={{ margin: "0 0 24px 0" }}>
                <MinusCircleOutlined
                  onClick={() => {
                    onFieldRemove(field.name);
                    remove(field.name);
                  }}
                />
              </Col>
            </Row>
          ))}

          <Button
            type="dashed"
            onClick={() => add()}
            block
            icon={<PlusOutlined />}
          >
            Add step
          </Button>
        </>
      )}
    </Form.List>
  </>
);

export default LoadConfigurationForm;
