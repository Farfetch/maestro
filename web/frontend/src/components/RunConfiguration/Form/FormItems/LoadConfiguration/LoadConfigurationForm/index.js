import LoadConfigurationForm from "./LoadConfigurationForm";

export default LoadConfigurationForm;
