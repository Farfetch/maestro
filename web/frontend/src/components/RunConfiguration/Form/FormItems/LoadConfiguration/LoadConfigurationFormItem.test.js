import { cleanup, render, waitFor } from "@testing-library/react";
import { Form } from "antd";
import React from "react";

import LoadConfigurationForm from "./LoadConfigurationForm/LoadConfigurationForm";

afterEach(cleanup);

describe("components/RunConfiguration/Form/FormItems/LoadConfigurationForm", () => {
  test(`should render LoadConfigurationForm component`, async () => {
    const { container } = render(
      <Form>
        <LoadConfigurationForm />
      </Form>
    );

    await waitFor(() => {
      expect(container.outerHTML).toMatchSnapshot();
    });
  });
});
