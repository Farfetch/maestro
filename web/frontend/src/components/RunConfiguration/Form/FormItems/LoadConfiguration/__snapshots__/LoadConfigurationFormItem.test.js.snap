// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/RunConfiguration/Form/FormItems/LoadConfigurationForm should render LoadConfigurationForm component 1`] = `"<div><form class=\\"ant-form ant-form-horizontal\\"><button type=\\"button\\" class=\\"ant-btn ant-btn-dashed ant-btn-block\\"><span role=\\"img\\" aria-label=\\"plus\\" class=\\"anticon anticon-plus\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"plus\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><defs><style></style></defs><path d=\\"M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z\\"></path><path d=\\"M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z\\"></path></svg></span><span>Add step</span></button></form></div>"`;
